src/StoreApi/Routes/Checkout.php

```diff
@@ -164,6 +164,32 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		// Check order is still valid.
 		$order_controller->validate_order_before_payment( $order_object );
 
+		// Create a new user account as necessary.
+		if ( ! is_user_logged_in() && ( WC()->checkout()->is_registration_required() || ! empty( $request['create_account'] ) ) ) {
+			$username    = '';
```

Comment:
Why are these variables needed?

Reply:
Work in progress - these will be gone, thanks for highlighting!